app/jobs/scheduled/clean_up_tags.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Jobs
+  class CleanUpTags < ::Jobs::Scheduled
+    every 24.hours
+
+    def execute(args)
+      return unless SiteSetting.automatically_clean_tags
+      Tag.unused.destroy_all
```

Comment:
I see what the `unused` scope does, it relies on counts. I wonder if there is a potential for these counts to be out of sync with reality? We may end up deleting tags that are actually still attached to a topic. Could you please check that?

Reply:
Very good question. Counts are updated on `after_create` callback for `TopicTag`:

https://github.com/discourse/discourse/blob/main/app/models/topic_tag.rb#L18

The only but unlikely risk is that someone create a topic with new tag at the moment when job is doing cleanup. 

I added grace period for newly created tags to avoid that situation.

Reply:
Nice, sounds good :)